src/python/bcc/table.py

```diff
@@ -922,3 +947,48 @@ def __init__(self, *args, **kwargs):
 class MapInMapHash(HashTable):
     def __init__(self, *args, **kwargs):
         super(MapInMapHash, self).__init__(*args, **kwargs)
+
+class RingBuf(EventArrayBase):
+
+    def __init__(self, *args, **kwargs):
+        super(RingBuf, self).__init__(*args, **kwargs)
+
+    def __delitem(self, key):
+        return
+
+    def __del__(self):
+        lib.bpf_free_ringbuf(self.ringbuf)
+        del self.bpf.ring_buffers[id(self)]
+
+    def open_ring_buffer(self, callback):
+        """open_ring_buffer(callback)
+
+        Opens a ring buffer to receive custom event data from the bpf program.
+        The callback will be invoked for each event submitted from the kernel,
+        up to millions per second.
+        """
+
+        def ringbuf_cb_(ctx, data, size):
+            try:
+                ret = callback(ctx, data, size)
+                # Callback for ringbufs should _always_ return an integer.
+                # If the function the user registers does not,
+                # simply fall back to returning 0.
+                try:
+                    ret = int(ret)
+                except:
+                    ret = 0
+            except IOError as e:
+                if e.errno == errno.EPIPE:
+                    exit()
+                else:
+                    raise e
+            return ret
+
+        fn = _RINGBUF_CB_TYPE(ringbuf_cb_)
+        self.ringbuf = lib.bpf_new_ringbuf(self.map_fd, fn)
+        if not self.ringbuf:
+            raise Exception("Could not open ring buffer")
+        self.bpf.ring_buffers[id(self)] = self.ringbuf
```

Comment:
libbpf's ring_buffer is already a manager of many ring buffers. It would be simpler to just use that, instead of maintaining list of ringbuffers on Python side? Unless I'm missing something, not an expert on BCC internals.

Reply:
I chose to do it this way to make it more similar to perf buffer implementation. But I'm open to changing this.

Reply:
I specifically designed ring_buffer in libbpf to be able to handle multiple ring buffer maps simultaneously within a single epoll instance. I saw that in practice some applications have multiple perfbuf/ringbufs and they are going in loop to poll them, which is very suboptimal. So while we don't have much choice for perf_buffer, please do better for ring_buffers.

Reply:
The performance argument makes a lot of sense. This will represent a significant change compared to how we handle other map types, but it is probably for the best. I'll try and get this working as soon as I can.

Reply:
Latency as well, not just performance :)